### ISSUE TITLE
Google chrome autofill is displayed over autocomplete results

### ISSUE THREAD

**Issue by gregor2004ua** [ErrorPro/react-google-autocomplete#34]
Cant get chrome autofill to disappear:
![image](https://user-images.githubusercontent.com/4721049/43361962-d13d0af4-92ac-11e8-8946-24b83b640378.png)

Here is what the code looks like. I am using it with redux-form and autofill three hidden fields based on selection.
```
<Autocomplete
        onPlaceSelected={(place) => {
          autofill('address', place.formatted_address);
          autofill('coordinates.latitude', place.geometry.location.lat().toString());
          autofill('coordinates.longitude', place.geometry.location.lng().toString());
        }}
        types={['establishment', 'geocode']}
        className="form-control"
        placeholder="Number Street, City, State, Country"
        autoComplete="address-autocomplete"
        {...input}
        onKeyPress={e => {
          if (e.key === 'Enter') e.preventDefault();
        }}
      />
```

any help would be greatly appreciated.

**Comment by ErrorPro** [comment]
@gregor2004ua Thanks for the issue, will check it asap.

**Comment by herebefrogs** [comment]
@gregor2004ua I ran into this issue too. The root cause is that Chrome Autofill ignores the `autocomplete="off"` attribute that the Google Maps library add to the input generated by ReactGoogleAutocomplete, which is quite unfortunate given Google manufactures both the browser and library.

The workaround is to alter the `autocomplete` value on the input to a key that does not match anything in the Autofill address object (e.g. "new-password" or "nope" will do) so Autofill, even though still active, can't find anything to pre-fill the `address` field with.

Given the Google Maps library doesn't provide any config option to set the `autocomplete` value (https://github.com/ErrorPro/react-google-autocomplete/blob/master/src/index.js#L29) and that it will overwrite whatever `autocomplete` prop we might pass we might pass to ReactGoogleAutocomplete (https://github.com/ErrorPro/react-google-autocomplete/blob/master/src/index.js#L50), I have implemented this solution from Stackoverflow (listen for DOM mutations on the input to detect the the addition of `autocomplete="off"` and immediately overwrite it with a different value of our own): https://stackoverflow.com/questions/29931712/chrome-autofill-covers-autocomplete-for-google-maps-api-v3/49161445#49161445

I verified this stops Chrome and Edge's Autofill, but not Safari's. Firefox's Autofill doesn't cause this problem in the first place since it only offers to autofill logins and passwords. Hope this helps.

@ErrorPro if you're interested, I could work on a Pull Request to integrate that MutationObserver directly inside ReactGoogleAutocomplete. Let me know what you think.
